apps/channels/forms.py

```diff
@@ -71,6 +71,33 @@ def get_success_message(self, channel: ExperimentChannel):
         return f"Use the following URL when setting up the webhook: {channel.webhook_url}"
 
 
+class InAppChannelForm(ExtraFormBase):
+    client_id = forms.CharField(label="Client ID", max_length=100)
+    webook_url = forms.CharField(
+        widget=forms.TextInput(attrs={"readonly": "readonly"}),
+        label="Webhook URL",
+        disabled=True,
+        required=False,
+        help_text="Use this as the URL when setting up the webhook",
+    )
+
+    def __init__(self, *args, **kwargs):
+        initial = kwargs.get("initial", {})
+        channel: ExperimentChannel = kwargs.pop("channel", None)
+        if channel:
+            initial["webook_url"] = channel.webhook_url
+            kwargs["initial"] = initial
+
+        super().__init__(*args, **kwargs)
+        if not channel:
+            # We only show the webhook URL field when there is something to show
+            self.fields["webook_url"].widget = forms.HiddenInput()
+
+    def get_success_message(self, channel: ExperimentChannel):
+        """The message to be displayed when the channel is successfully linked"""
+        return f"Use the following URL when setting up the webhook: {channel.webhook_url}"
```

Comment:
> The get_success_message method, remains in the respective classes. Let me know if you think we should move it to superclass as well.

I think it should also move since at the moment all the messages are identical. If a class wants to change the message it can override that method.